devicetypes/smartthings/zwave-metering-dimmer.src/zwave-metering-dimmer.groovy

```diff
@@ -209,17 +222,138 @@ def ping() {
 }
 
 def refresh() {
-	delayBetween([
-		zwave.switchMultilevelV1.switchMultilevelGet().format(),
-		zwave.meterV2.meterGet(scale: 0).format(),
-		zwave.meterV2.meterGet(scale: 2).format(),
-	], 1000)
+    log.debug "refresh()"
+
+    encapSequence([
+        zwave.switchMultilevelV1.switchMultilevelGet(),
+        meterGet(scale: 0),
+        meterGet(scale: 2),
+    ], 1000)
 }
 
 def setLevel(level) {
 	if(level > 99) level = 99
-	delayBetween([
-		zwave.basicV1.basicSet(value: level).format(),
-		zwave.switchMultilevelV1.switchMultilevelGet().format()
+	encapSequence([
+		zwave.basicV1.basicSet(value: level),
+		zwave.switchMultilevelV1.switchMultilevelGet()
 	], 5000)
 }
+
+def configure() {
+	log.debug "configure()"
+    def result = []
+
+    log.debug "Configure zwaveInfo: "+zwaveInfo
+
+    if (zwaveInfo.mfr == "0086") {  // Aeon Labs meter
+    	// TEST CONFIGS
+		result << response(encap(zwave.configurationV1.configurationSet(parameterNumber: 90, size: 1, scaledConfigurationValue: 1))) // enable threashhold values for wattage reporting
+		result << response(encap(zwave.configurationV1.configurationSet(parameterNumber: 91, size: 2, scaledConfigurationValue: 0))) // min change in wattage
+		result << response(encap(zwave.configurationV1.configurationSet(parameterNumber: 92, size: 1, scaledConfigurationValue: 0))) // min change in wattage %
+		// END OF TEST CONFIGS
+
+		result << response(encap(zwave.configurationV1.configurationSet(parameterNumber: 80, size: 1, scaledConfigurationValue: 2)))    // basic report cc
+		result << response(encap(zwave.configurationV1.configurationSet(parameterNumber: 101, size: 4, scaledConfigurationValue: 4)))   // report power in watts
+		result << response(encap(zwave.configurationV1.configurationSet(parameterNumber: 111, size: 4, scaledConfigurationValue: 300)))  // every 5 min
+		result << response(encap(zwave.configurationV1.configurationSet(parameterNumber: 102, size: 4, scaledConfigurationValue: 8)))   // report energy in kWh
+		result << response(encap(zwave.configurationV1.configurationSet(parameterNumber: 112, size: 4, scaledConfigurationValue: 300)))  // every 5 min
+		result << response(encap(zwave.configurationV1.configurationSet(parameterNumber: 103, size: 4, scaledConfigurationValue: 0)))   // no third report
+		result << response(encap(zwave.configurationV1.configurationSet(parameterNumber: 113, size: 4, scaledConfigurationValue: 300))) // every 5 min
+ 	}
+	result << response(encap(meterGet(scale: 0)))
+	result << response(encap(meterGet(scale: 2)))
+}
+
+def reset() {
+	encapSequence([
+		meterReset(),
+		meterGet(scale: 0)
+	])
+}
+
+def meterGet(scale)
+{
+    zwave.meterV2.meterGet(scale)
+}
+
+def meterReset()
+{
+    zwave.meterV2.meterReset()
+}
+
+def isNano()
+{
+    return (zwaveInfo.mfr == "0086" && zwaveInfo.prod == "0103")
+}
+
+def normalizeLevel(level)
+{
+    (int)(Math.floor((float)level * 1.01010101))
```

Comment:
@mckeed @ALamchaS : Here's the same problem again.

I think this is functionally equivalent to just setting 99 to 100. It looks a bit less hacky, but it is a little more math.

Reply:
Math.floor is redundant here as casting to int does the same.

If-else or ? : makes this code easier to understand. I was not certain level 99 will result in 100 until I tested this. If not, a comment what this line does should be added.

There should not be end of line before '{'